### PR TITLE
BlockchainUpdates API extension

### PR DESCRIPTION
**Pull request by dvshur**

### CODE REVIEW

File: blockchain-updates/build.sbt [hunk 1]
```diff
@@ -2,25 +2,33 @@ import WavesDockerKeys._
...
 
 inConfig(Compile)(
   Seq(
-    PB.protoSources in Compile := Seq(PB.externalIncludePath.value),
-    includeFilter in PB.generate := new SimpleFileFilter((f: File) => f.getName.endsWith(".proto") && f.getParent.replace('\\', '/').endsWith("waves/events")),
+    PB.protoSources in Compile := Seq(PB.externalIncludePath.value, sourceDirectory.value / "protobuf"),
+    includeFilter in PB.generate := new SimpleFileFilter(
+      (f: File) => {
+        val name = f.getName
+        name == "blockchain_updates.proto" || name == "events_temp.proto" || (name.endsWith(".proto") && f.getParent
```
Comment (dvshur) [inline blockchain-updates/build.sbt:15]: These files will be moved to `protobuf-schemas` before production.

File: build.sbt [hunk 2]
```diff
@@ -112,7 +112,8 @@ inScope(Global)(
       "-Ywarn-unused:-implicits",
       "-Xlint",
       "-opt:l:inline",
-      "-opt-inline-from:**"
+      "-opt-inline-from:**",
+      "-target:jvm-1.8"
```
Comment (vsuharnikov) [inline build.sbt:116]: Why do we need this option?

File: blockchain-updates/src/main/protobuf/events_temp.proto [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        oneof body {
+            BlockAppend block = 1;
+            MicroBlockAppend micro_block = 2;
+        }
+
+        repeated bytes transaction_ids = 3;
+
+        StateUpdate state_update = 11;
+        repeated StateUpdate transaction_state_updates = 12;
```
[the remark this answers is not in the document]
Comment (dvshur) [reply blockchain-updates/src/main/protobuf/events_temp.proto:32]: Yes, it does

File: build.sbt (hunk 2 above)
Comment (dvshur) [reply build.sbt:116]: Fixed a deploy problem for us.

File: blockchain-updates/src/main/protobuf/events_temp.proto [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+        bool asset_existed_before = 20;
+        /*
+                Related to a past behavior in the blockchain when it was possible to reissue
+                assets so that the total volume became more then int64 max value.
+                This field represents accurate volume even for those assets.
+                Can be ignored if the target system does not need such accuracy.
+                Encoding: like Java BigInteger, https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html#toByteArray()
+            */
+        bytes safe_volume = 21;
```
[the remark this answers is not in the document]
Comment (dvshur) [reply blockchain-updates/src/main/protobuf/events_temp.proto:96]: I'm not sure if Amount is used in this context. Maybe Amount represents a single operation? For any single operation int64 is limit.

**Pull request merged**